x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/utils.test.ts

```diff
@@ -0,0 +1,658 @@
+        mockLogger
+      );
+      expect(mockLogger.warn).toHaveBeenCalledTimes(1);
+      expect(result).toEqual(false);
+    });
+    test('if unsuccessful iteration of searchAfterAndBulkIndex due to empty sort ids and 0 total hits', async () => {
+      const searchAfterSortId = '1234567891111';
+      const sampleParams = sampleSignalAlertParams(undefined);
+      const savedObjectsClient = savedObjectsClientMock.create();
+      const expectedSearchAfterQuery = buildEventsSearchQuery({
+        index: sampleParams.index,
+        from: sampleParams.from,
+        to: sampleParams.to,
+        filter: sampleParams.filter,
+        size: sampleParams.size ? sampleParams.size : 1,
+        searchAfterSortId,
+        maxDocs: undefined,
+      });
+      const mockService: AlertServices = {
+        callCluster: async (action: string, params: SearchParams) => {
+          if (action === 'bulk') {
+            expect(action).toEqual('bulk');
+            expect(params.index).toEqual('.siem-signals-10-01-2019');
+            return {
+              took: 100,
+              errors: false,
+              items: [
+                {
+                  fakeItemValue: 'fakeItemKey',
+                },
+              ],
+            };
+          }
+          expect(action).toEqual('search');
+          expect(params.index).toEqual(sampleParams.index);
+          expect(params).toEqual(expectedSearchAfterQuery);
+          return sampleDocSearchResultsWithSortId;
+        },
+        alertInstanceFactory: jest.fn(),
+        savedObjectsClient,
+      };
+      const result = await searchAfterAndBulkIndex(
+        sampleDocSearchResultsNoSortIdNoHits,
+        sampleParams,
+        mockService,
+        mockLogger
+      );
+      expect(result).toEqual(true);
+    });
+    test('if successful iteration of while loop with maxDocs and search after returns results with no sort ids', async () => {
+      const searchAfterSortId = '1234567891111';
+      const sampleParams = sampleSignalAlertParams(10);
+      const savedObjectsClient = savedObjectsClientMock.create();
+      const expectedSearchAfterQuery = buildEventsSearchQuery({
+        index: sampleParams.index,
+        from: sampleParams.from,
+        to: sampleParams.to,
+        filter: sampleParams.filter,
+        size: sampleParams.size ? sampleParams.size : 1,
+        searchAfterSortId,
+        maxDocs: sampleParams.maxSignals,
+      });
+      const mockService: AlertServices = {
+        callCluster: async (action: string, params: SearchParams) => {
+          if (action === 'bulk') {
+            expect(action).toEqual('bulk');
+            expect(params.index).toEqual('.siem-signals-10-01-2019');
+            return {
+              took: 100,
+              errors: false,
+              items: [
+                {
+                  fakeItemValue: 'fakeItemKey',
+                },
+              ],
+            };
+          }
+          expect(action).toEqual('search');
```

Comment:
Same comment as above, starting on line 538-540, these expect statements look to never reached. I would remove them and the if check if that is the case.